zerver/lib/tex.py

```diff
@@ -0,0 +1,43 @@
+import os
+import subprocess
+from django.conf import settings
+from typing import Text
+from zerver.lib.str_utils import force_bytes
+
+def render_tex(tex, is_inline=True):
+    # type: (Text, bool) -> Text
+    """Render a TeX string into HTML using KaTeX
+
+    Returns the HTML string, or None if there was some error in the TeX syntax
+
+    Keyword arguments:
+    tex -- Text string with the TeX to render
+           Don't include delimiters ('$$', '\[ \]', etc.)
+    is_inline -- Boolean setting that indicates whether the render should be
+                 inline (i.e. for embedding it in text) or not. The latter
+                 will show the content centered, and in the "expanded" form
+                 (default True)
+    """
+
+    command = ['node']
+    if os.path.exists(os.path.join(settings.DEPLOY_ROOT, 'prod-static/')):
+        # Running in production
+        command.append(os.path.join(settings.DEPLOY_ROOT,
+                                    'prod-static/serve/third/katex/cli.js'))
```

Comment:
This would be cleaner as `os.path.join(settings.STATIC_ROOT, 'third/katex/cli.jsthird/katex/cli.js')`